File: packages/shared/src/models/return-request.ts
export interface ReturnItem {
  ean: string;
  qty: number;
  unit_price: number;
  reason: string;
}

export type RefundMethod = "loyalty_credit" | "card_refund";
export type RefundStatus = "pending" | "approved" | "rejected" | "processed";

export interface Refund {
  amount: number;
  method: RefundMethod;
  status: RefundStatus;
}

export interface ReturnRequest {
  type: "return_request";
  return_id: string;
  user_id: string;
  store_id: string;
  original_txn_id: string;
  items: ReturnItem[];
  refund: Refund;
  points_delta: number;
  offline: boolean;
  created_at: string;
}

File: packages/server/app/models/return_request.py
from typing import Literal

from pydantic import BaseModel

RefundMethod = Literal["loyalty_credit", "card_refund"]
RefundStatus = Literal["pending", "approved", "rejected", "processed"]


class ReturnItem(BaseModel):
    ean: str
    qty: int
    unit_price: float
    reason: str


class Refund(BaseModel):
    amount: float
    method: RefundMethod
    status: RefundStatus = "pending"


class ReturnRequest(BaseModel):
    type: Literal["return_request"] = "return_request"
    return_id: str
    user_id: str
    store_id: str
    original_txn_id: str
    items: list[ReturnItem] = []
    refund: Refund
    points_delta: int
    offline: bool = False
    created_at: str
